File: LeetCode - May/firstUniqueChar.java
class Solution {
    public int firstUniqChar(String s) {
        int result = -1;
        HashMap<Character, Integer> counter = new HashMap<>();

        for(int x = 0; x < s.length(); x++){
            counter.put(s.charAt(x), counter.getOrDefault(s.charAt(x),0) + 1);
        }

        for (int x = 0; x < s.length(); x++) {
            char i = s.charAt(x);
            if(counter.get(i) == 1){
                return s.indexOf(i); 
            }
        }

        return -1;
    }
}

File: LeetCode - May/firstUniqueChar.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        counter={}
        for i in s:
            counter[i]=counter.get(i,0)+1
        
        for i in s:
            if(counter[i]==1):
                return s.index(i)
        return -1
            

        